scalyr_agent/scalyr_client.py

```diff
@@ -1246,16 +1299,19 @@ def __set_attributes( self, thread_id, attributes ):
             tmp_buffer.write('thread:')
             tmp_buffer.write( scalyr_util.json_encode(thread_id) )
             tmp_buffer.write(', ')
-        if attributes is not None:
+            tmp_buffer.write('log:')
+            tmp_buffer.write( scalyr_util.json_encode(thread_id) )
+            tmp_buffer.write(', ')
+        #if attributes is not None:
             # Serialize the attributes object, but we have to remove the closing brace because we want to
             # insert more fields.
-            tmp_buffer.write('attrs:')
-            tmp_buffer.write( scalyr_util.json_encode(attributes) )
-            _rewind_past_close_curly(tmp_buffer)
-            tmp_buffer.write(',')
-        else:
+        #    tmp_buffer.write('attrs:')
```

Comment:
Should this really be deleted?

Reply:
That is the part that inserts all the log attributes into the single event, so if we want to reduce the message size this is what we remove. Though I shouldn't have left it commented out, should have just removed.

Reply:
I was thinking we should probably put the entire feature behind a configuration parameter - on by default, but can be turned off in case there are problems.

Reply:
That's probably a good idea.

Reply:
I debated this in my head a bit (and just had a discussion with Yan).  I would say no, we don't add in this flag.  The main reason I could see us doing it is if we wanted to disable the support in the servers and have customers be able to react to that without upgrading the agent.

However, once this API is public, we cannot control if a customer is using it or not.  So, the server must always support it going forward.

The only thing the flag would give us is a way to turn it off if there is a bug in the client code.  Well, I don't think we should have flags just for that.. if there was a bug (and we do have testing already) then we would either quickly roll out a new version or tell customers to roll back.

Reply:
> The only thing the flag would give us is a way to turn it off if there is a bug in the client code.

Or a bug in the server code.

Reply:
Since the server hasn't rolled this out everywhere, ok, let's go ahead and add in the configuration option with a comment saying this will be deleted at some point in the future.